### PR TITLE
feat(auth-server): add paypal subscription create route

### LINKED ISSUE #7326: Create paypal handler for customer authorization
A Paypal handler should be created for /oauth/subscription/active/new-paypal per the linked issue references. This will make a variety of calls per:

[https://github.com/mozilla/ecosystem-platform/blob/master/docs/fxa-engineering/fxa-sub-platform.md#paypal-checkout](https://github.com/mozilla/ecosystem-platform/blob/master/docs/fxa-engineering/fxa-sub-platform.md#paypal-checkout)

It will need to make multiple stripe helper and paypal helper calls to store references, create billing agreement, etc.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-2938)
┆Issue Number: FXA-2938

### PR DESCRIPTION
**Pull request by bbangert**
Because:

* We want to handle creating a subscription and paying it via PayPal
  when a user chooses PayPal.

This commit:

* Adds a paypal-new subscription route that will create a billing
  agreement, subscription, and setup the Stripe data for recurring
  billing to PayPal.

Closes #7326

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: packages/fxa-auth-server/lib/routes/subscriptions/paypal.ts [hunk 1]
```diff
@@ -82,5 +162,30 @@ export const paypalRoutes = (
...
+        auth: {
+          payload: false,
+          strategy: 'oauthToken',
+        },
+        response: {
+          schema: isA.object().keys({
+            subscription: validators.subscriptionsStripeSubscriptionValidator,
+            sourceCountry: validators.subscriptionPaymentCountryCode.required(),
+          }),
+        },
```
Comment (biancadanforth) [inline packages/fxa-auth-server/lib/routes/subscriptions/paypal.ts:178]: The validation here still throws an error and returns a 500 response for me on the client. I only get a 200 when I add `failAction: 'log',` as we had to do for the demo. Can we fix this before merging else file a follow-up to fix it next?

**Review by biancadanforth — CHANGES_REQUESTED** [review]
Thanks Ben for this PR.

I haven't looked at everything closely, but I just ran the end-to-end checkout flow on the `fxa-payments-server`, and I got back a `500` response again. It's the same validation issue on the `'/oauth/subscriptions/active/new-paypal'` as we ran into for the demo that doesn't show up in the auth server logs. See my comment in-line.

**Review by biancadanforth — APPROVED** [review]

File: packages/fxa-auth-server/lib/payments/stripe.ts [hunk 2]
```diff
@@ -557,6 +688,30 @@ export class StripeHelper {
...
+      throw error.internalValidationError(
+        'findCustomerSubscriptionByPlanId',
+        {
+          customerId: customer.id,
+        },
+        'Expected subscriptions to be loaded.'
+      );
+    }
+    return customer.subscriptions.data.find(
+      (sub) => sub.items.data.find((item) => item.plan.id === planId) != null
```
Comment (chenba) [inline packages/fxa-auth-server/lib/payments/stripe.ts:711]: nit: `sub.items.data.some` would read better here.

Comment (bbangert) [reply packages/fxa-auth-server/lib/payments/stripe.ts:711]: This method was just moved around, though wouldn't `some` get a list while this wants a single item?

Comment (chenba) [reply packages/fxa-auth-server/lib/payments/stripe.ts:711]: `some` just help with removing the ` != null` comparison as it already returns a boolean.  Like I said, nit. :D

**Review by chenba — APPROVED** [review]
LGTM

**Pull request merged**